syntax= "proto3";

package spectrogramservice;

service SpectrogramService {
    rpc StreamSpectrogram (stream SpectrogramRequest) returns (SpectrogramData);
}

message SpectrogramRequest {
    bytes audio_data = 1;
    optional SpectrogramParams params = 2;
}

message SpectrogramParams {
  string window_type = 1;
  int32 n_segment = 2;
  int32 highpass_cutoff = 3;
  float color_scale_min = 4;
  int32 max_displayed_frequency = 5;
}

message SpectrogramData {
    repeated float frequencies = 1;
    repeated float times = 2;
    repeated float power_levels = 3;
}